tests/isomorphism/setup-and-markup/Counter.tsx

```diff
@@ -0,0 +1,23 @@
+/** @jsx jsx */
+import { Component } from 'xoid'
+
+const Counter: Component<{ initialValue: number }> = ($props, { effect }) => {
+  const $counter = $props.map((s) => s.initialValue)
+  const increment = () => $counter.update((s) => s + 1)
+  const decrement = () => $counter.update((s) => s - 1)
+
+  effect(() => {
+    console.log('mounted')
+    return () => console.log('unmounted')
+  })
+
+  return (get, jsx) => (
+    <div>
+      count: {get($counter)}
```

Comment:
On the topic of implicit dependencies that I brought up in [this comment](https://github.com/onurkerimov/xoid/pull/17#issuecomment-1550203159), I'd worry this will be an easy place for devs to make a mistake forgetting to use `get($counter)` as opposed to `$counter.value` (or `$counter.get()`) just working correctly if we had implicit dependency tracking. 😀

Also, if we had implicit dependency tracking, we could make the `effect` callback track dependencies as well to work much like Solid and Preact Signals, which would be really nice!

Reply:
I really value this opinion. However, I'd also like to keep the core library's stance as "always explicit / no magical subscriptions". Perhaps xoid can have a `@xoid/auto` module (just like how MobX names it):

```js
import auto from '@xoid/auto'

// watch 
auto(() => {
  console.log('auto updates by .value:', $counter.value)
})

// derived state 
const $doubleCounter = auto(() => 2 * $counter.value)
```